sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,297 @@ def reduce_inequalities(inequalities, symbols=[]):
+            eqns.append( rel.rhs - rel.lhs)
+        elif type(rel) == Eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+        else:
+            raise TypeError(f"Unrecognized relational: {rel}")
+
+    variables = sorted(variables, key=lambda v: str(v)) # order symbols lexicographically
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(objective, *variables)[0] # constant terms can be safely ignored here
+
+    return A, B, C
+
+# Maximize Cx + D constrained with Ax <= B and x >= 0
+# Minimize y^{T}B constrained with y^{T}A >= C^{T} and y >= 0
+def linear_programming(constraints, objective):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method.
+
+    Explanation
+    ===========
+
+    When x is a column vector of variables, y is a column vector of dual variables,
+    and when the objective is either:
+
+    *) Maximizing Cx constrained to Ax <= B and x >= 0.
+    *) Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    The method returns a triplet of solutions optimum, argmax and argmax_dual where
+    optimum is the optimum solution, and argmax/ argmax_dual give the
+    values for x and y, respectively.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        The lhs of these inequalities are equivalent to the matrix A and the rhs
+        are equivalent to the column vector B.
+
+    objective : a linear function to maximize
+        The objective function is represented by the row vector C.
+
+    random_seed : int
+        Determines the random pivoting behavior of the simplex method
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linear_programming
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+5*z)
+    >>> optimum
+    11/3
+    >>> argmax
+    [0, 1/3, 2/3]
+    >>> argmax_dual
+    [0, 2/3, 1]
+
+    The method also works with non-integer coefficients (like sqrt(2)).
+
+    >>> from sympy import sqrt
+    >>> r1 = y + sqrt(2)*z <= sqrt(3)
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+sqrt(5)*z)
+    >>> optimum
+    3
+    >>> argmax
+    [2, 1, 0]
+    >>> argmax_dual
+    [0, 1, 1]
+
+    See Also
+    ========
+
+    solve_univariate_inequality
+    """
+    from sympy.matrices.dense import Matrix
+    A, B, C = _to_standard_form(constraints, objective)
+    D = ImmutableMatrix([0])
+    M = Matrix([[A, B], [-C, D]])
+    r_orig = ['x_{}'.format(j) for j in range(M.cols - 1)]
+    s_orig = ['y_{}'.format(i) for i in range(M.rows - 1)]
+
+    r = r_orig.copy()
+    s = s_orig.copy()
+
+    M, r, s = _simplex(M, r, s)
```

Comment:
I don't think it's necessary to pass in `r` and `s` here. The `_simplex` function can just make a list of integers and permute it then it would return something like:
```
r = [2, 3, 1, 0]
```
Then below you can just do something like:
```
argmax = [M[ri, -1] for ri in r]
```

_Originally posted by @oscarbenjamin in https://github.com/sympy/sympy/pull/21687#discussion_r703022556_

Reply:
I don't think this would work. A variable that was originally associated with a column may later become associated with a row. If it returned something like
```
r = [2, 3, 1, 0]
```
then it wouldn't be clear whether `2` was originally a column or row variable. 

Here's an example of a pivot operation which shows how a variable associated with a row may switch places with a variable associated with a column. (The image is from the document cited in the docstring of the simplex method).
<img width="377" alt="Screen Shot 2023-06-26 at 5 35 13 PM" src="https://github.com/sympy/sympy/assets/67246777/afba39fd-8038-4a81-b0b6-f53b757988de">

Let me know if I'm misunderstanding something.